mmv1/third_party/terraform/tests/resource_firestore_field_test.go

```diff
@@ -0,0 +1,135 @@
+}
+
+func testAccFirestoreField_runUpdateTest(updateConfig string, t *testing.T, context map[string]interface{}) {
+	resourceName := context["resource_name"].(string)
+
+	VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckFirestoreFieldDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccFirestoreField_firestoreFieldUpdateInitialExample(context),
+			},
+			{
+				ResourceName:            fmt.Sprintf("google_firestore_field.%s", resourceName),
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"database", "collection", "field"},
+			},
+			{
+				Config: updateConfig,
+			},
+			{
+				ResourceName:            fmt.Sprintf("google_firestore_field.%s", resourceName),
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"database", "collection", "field"},
+			},
+			{
+				Config: testAccFirestoreField_firestoreFieldUpdateInitialExample(context),
+			},
+			{
+				ResourceName:            fmt.Sprintf("google_firestore_field.%s", resourceName),
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"database", "collection", "field"},
+			},
+		},
+	})
+}
+
+func testAccFirestoreField_firestoreFieldUpdateInitialExample(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_firestore_field" "%{resource_name}" {
+  project = "%{project_id}"
+  collection = "chatrooms_%{random_suffix}"
+  field = "%{resource_name}"
+
+  index_config {
+    indexes {
+        order = "ASCENDING"
+        query_scope = "COLLECTION_GROUP"
+    }
+    indexes {
+        array_config = "CONTAINS"
+    }
+  }
+}
+`, context)
+}
+
+func testAccFirestoreField_firestoreFieldUpdateAddTTLExample(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_firestore_field" "%{resource_name}" {
+  project = "%{project_id}"
+  collection = "chatrooms_%{random_suffix}"
+  field = "%{resource_name}"
+
+  index_config {
+    indexes {
+        order = "ASCENDING"
+        query_scope = "COLLECTION_GROUP"
+    }
+    indexes {
+        array_config = "CONTAINS"
+    }
+  }
+
+  ttl_config {}
+}
+`, context)
+}
+
+func testAccFirestoreField_firestoreFieldUpdateAddIndexExample(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_firestore_field" "%{resource_name}" {
+  project = "%{project_id}"
+  collection = "chatrooms_%{random_suffix}"
+  field = "%{resource_name}"
+
+  index_config {
+    indexes {
+        order = "ASCENDING"
+        query_scope = "COLLECTION_GROUP"
+    }
+    indexes {
+        array_config = "CONTAINS"
+    }
+	indexes {
+		order = "DESCENDING"
+		query_scope = "COLLECTION_GROUP"
+	}
```

Comment:
```suggestion
    indexes {
        array_config = "CONTAINS"
    }
    indexes {
        order = "DESCENDING"
        query_scope = "COLLECTION_GROUP"
    }
```

A very minor change, but not a blocker to merging as the downstream Go code is formatted during its generation

Reply:
To clarify, these should be tabs? I made that change (but am having a hard time knowing if it was the right thing to do as it's hard to see the difference in the comment).

Reply:
Sorry, I didn't see this comment 2 days ago. I don't think we're strict about tabs vs spaces, but in this case the varying indentation made it look like the second indexes block was nested inside another block.